File: LeetCode/Easy/2006. Count Number of Pairs With Absolute Difference K/2024-01-28_Accepted.cpp
class Solution {
public:
  int countKDifference(vector<int>& nums, int k) {
    int ans = 0;
    map<int, int> count;
    for (auto num : nums) {
      if (count.find(num) == count.end()) {
        count[num] = 0;
      }
      ans += getFromMap(count, num - k) + getFromMap(count, num + k);
      count[num]++;
    }
    return ans;
  }

private:
  int getFromMap(map<int, int>& count, int num) {
    if (count.find(num) != count.end()) {
      return count[num];
    }
    return 0;
  }
};

File: LeetCode/Easy/2085. Count Common Words With One Occurrence/2024-01-28_Accepted.py
class Solution:
  def countWords(self, words1: List[str], words2: List[str]) -> int:
    count = {}
    for word in words1:
      if word not in count:
        count[word] = 1
      else:
        count[word] = 1001
    for word in words2:
      if word in count:
        count[word] -= 1
    ans = 0
    for word in count:
      if count[word] == 0:
        ans += 1
    return ans
